sdks/python/apache_beam/io/aws/s3io.py

```diff
@@ -112,10 +115,28 @@ def list_prefix(self, path, with_metadata=False):
       If ``with_metadata`` is False: dict of file name -> size; if
         ``with_metadata`` is True: dict of file name -> tuple(size, timestamp).
     """
+    file_info = {}
+    for file_metadata in self.list_files(path, with_metadata):
+      file_info[file_metadata[0]] = file_metadata[1]
+
+    return file_info
+
+  def list_files(self, path, with_metadata=False):
```

Comment:
Use a different name because the return type (generator of files) is different from list_prefix (dict of files). And now the latter is a wrapper for the former.